### PR TITLE
parser: update as option

### PR DESCRIPTION
**Pull request by siddontang**
select field alias can support back/double/single quote

fix https://github.com/pingcap/tidb/issues/323

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -2928,11 +2937,21 @@ TableIdentOpt:
...
-|	AsOpt 
+|	TableAsOpt 
 	{
 		$$ = $1
 	}
 
+TableAsOpt:
+	identifier
+	{
+		// TODO: check potential bug
```
Comment (qiuyesuifeng) [inline parser/parser.y]: What is this?

File: parser/parser.y [hunk 2]
```diff
@@ -2928,11 +2937,21 @@ TableIdentOpt:
...
-|	AsOpt 
+|	TableAsOpt 
 	{
 		$$ = $1
 	}
 
+TableAsOpt:
+	Identifier
+	{
+		// TODO: check potential bug
```
Comment (qiuyesuifeng) [inline parser/parser.y]: @shenli  Can u remember why we add this comment?

Comment (shenli) [inline parser/parser.y]: I am also confused....

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**